File: schema_salad/dotnet/util/Loaders/MapLoader.cs
using System.Collections;

namespace ${project_name};

internal class MapLoader<T> : ILoader<Dictionary<string, T>>
{
    private readonly ILoader valueLoader;

    public MapLoader(in ILoader valueLoader)
    {
        this.valueLoader = valueLoader;
    }

    public Dictionary<string, T> Load(in object doc, in string baseuri, in LoadingOptions loadingOptions, in string? docRoot = null)
    {
        if (doc == null)
        {
            throw new ValidationException("Expected non null");
        }

        if (doc is not IDictionary)
        {
            throw new ValidationException("Expected list");
        }

        IDictionary docDictionary = (IDictionary)doc;
        Dictionary<string, T> returnValue = new();
        List<ILoader> loaders = new()
        {
            this,
            valueLoader
        };
        ILoader<object> unionLoader = new UnionLoader(loaders);
        List<ValidationException> errors = new();

        foreach (KeyValuePair<string, T> item in docDictionary)
        {
            try
            {
                dynamic loadedField = unionLoader.LoadField(item.Value, baseuri, loadingOptions);
                returnValue[item.Key] = loadedField;
            }
            catch (ValidationException e)
            {
                errors.Add(e);
            }
        }

        if (errors.Count > 0)
        {
            throw new ValidationException("", errors);
        }

        return returnValue;
    }

    object ILoader.Load(in object doc, in string baseuri, in LoadingOptions loadingOptions, in string? docRoot)
    {
        return Load(doc, baseuri, loadingOptions, docRoot);
    }
}

File: schema_salad/typescript/util/loaders/MapLoader.ts
import {Loader, loadField, LoadingOptions, _UnionLoader, ValidationException, TypeGuards, Dictionary} from '../Internal'

export class _MapLoader implements Loader {
  values: Loader[]
  container?: string

  constructor (values: Loader[], container?: string) {
    this.values = values
    this.container = container
  }

  async load (doc: any, baseuri: string, loadingOptions: LoadingOptions, docRoot?: string): Promise<any> {
    if (!TypeGuards.isDictionary(doc)) {
      throw new ValidationException('Expected a dict')
    }
    if (this.container !== undefined) {
      loadingOptions = new LoadingOptions({ copyFrom: loadingOptions, container: this.container })
    }
    let r : Dictionary = {}
    const errors: ValidationException[] = []
    for (const key in doc) {
      try {
        r[key] = await loadField(doc[key], new _UnionLoader([this, ...this.values]), baseuri, loadingOptions)
      } catch (e) {
        if (e instanceof ValidationException) {
          errors.push(e)
        } else {
          throw e
        }
      }
    }
    
    if (errors.length > 0) {
      throw new ValidationException('', errors)
    }
    return r
  }
}
